modules/metastore/src/Reference/MetastoreUrlGenerator.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace Drupal\metastore\Reference;
+
+use Drupal\common\StreamWrapper\DkanStreamWrapper;
+use Drupal\Core\StreamWrapper\StreamWrapperManager;
+use Drupal\Core\StreamWrapper\StreamWrapperManagerInterface;
+use Drupal\metastore\Exception\MissingObjectException;
+use Drupal\metastore\MetastoreService;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Convert between local file paths and public file URLs.
+ *
+ * @todo Convert to service with Dependency Injection.
+ */
+class MetastoreUrlGenerator {
+  const DKAN_SCHEME = 'dkan';
+  const API_PATH = '/api/1';
+
+  /**
+   * DKAN Stream Wrapper.
+   *
+   * @var \Drupal\Core\StreamWrapper\StreamWrapperManager
+   */
+  protected StreamWrapperManager $streamWrapperManager;
+
+  /**
+   * Metastore service.
+   *
+   * @var \Drupal\metastore\MetastoreService
+   */
+  protected MetastoreService $metastore;
+
+  /**
+   * The request stack.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $requestStack;
+
+  /**
+   * Constructs a new file URL generator object.
+   *
+   * @param \Drupal\Core\StreamWrapper\StreamWrapperManagerInterface $stream_wrapper_manager
+   *   The stream wrapper manager.
+   * @param \Drupal\metastore\MetastoreService $metastore
+   *   Metastore service.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $request_stack
+   *   The request stack.
+   */
+  public function __construct(
+    StreamWrapperManagerInterface $stream_wrapper_manager,
+    MetastoreService $metastore,
+    RequestStack $request_stack
+  ) {
+    $this->streamWrapperManager = $stream_wrapper_manager;
+    $this->metastore = $metastore;
+    $this->requestStack = $request_stack;
+  }
+
+  /**
+   * Retrieve the metastore service.
+   *
+   * @return \Drupal\metastore\MetastoreService
+   *   Metastore service.
+   */
+  protected function metastore(): MetastoreService {
```

Comment:
Why do we need a method for this?

Reply:
Hmm.. I think that was to help make more testable, since properties can't be mocked, but seems unnecessary looking at it again...